cmd/registry.go

```diff
@@ -42,3 +50,35 @@ func AvailableCommands() []string {
 	sort.Strings(avail)
 	return avail
 }
+
+// Register a boulder config struct.
+func RegisterConfig(name string, f *ConfigValidator) {
```

Comment:
I don't see any locations where RegisterConfig and RegisterCommand aren't called in a pair. Why not just make `f *ConfigValidator` a third argument to `cmd.RegisterCommand()`? If there are places where we call RegisterCommand but not RegisterConfig, then we can just pass `nil` as the third argument there.

Reply:
> I don't see any locations where RegisterConfig and RegisterCommand aren't called in a pair. Why not just make f *ConfigValidator a third argument to cmd.RegisterCommand()? If there are places where we call RegisterCommand but not RegisterConfig, then we can just pass nil as the third argument there.

They're not called together fully 1/3 of the time. Locations where they're not called together:
  - cmd/boulder/main.go (2x)
  - cmd/caa-log-checker/main.go
  - cmd/crl-checker/main.go
  - cmd/id-exporter/main.go
  - cmd/log-validator/main.go
  - cmd/ceremony/main.go
  - cmd/orphan-finder/main.go
  - cmd/reversed-hostname-checker/main.go
  - cmd/rocsp-tool/main.go
  
I personally prefer that these remain separate but I could probably be convinced to combine them.

Reply:
It just feels to error prone -- in order to ensure that there's never a typo when registering the separate command and config you had to declare a bunch of `cmdName` consts in each main.go, and that feels like it shouldn't be necessary. Combinging them into a single `cmd.RegisterCommand("name", main, Config)` or `cmd.RegisterCommand("name", main, nil)` removes the need to even think about that.

Not every command has a config, but every config has a command. It should never be possible to register a config without registering the corresponding command. Therefore they should be registered at the same time, with only one nillable.

Reply:
Okay, I can agree with that. I'll go ahead and combine them.